api/src/classes/pagination.py

```diff
@@ -36,13 +41,35 @@ class CompanyPagination(Pagination):
 
     def __init__(
         self,
-        page: int = PaginationDefaults.PAGE.value,
-        limit: int = PaginationDefaults.LIMIT.value,
+        page: int = Check(default=PaginationDefaults.PAGE.value, ge=1),
+        limit: int = Check(default=PaginationDefaults.LIMIT.value, ge=1),
+        *,
+        ascending: bool = False,
+    ) -> None:
+        super().__init__(page=page, limit=limit)
+        self.ascending: bool | None = ascending
+
+    def as_dict(self) -> dict[str, Any]:
+        """Get the params as a dict."""
+        return super().as_dict()
+
+
+class ShopPagination(Pagination):
+    """Pagination for getting Shop Items."""
+
+    def __init__(
+        self,
+        page: int = Check(default=PaginationDefaults.PAGE.value, ge=1),
+        limit: int = Check(default=PaginationDefaults.LIMIT.value, ge=1),
         *,
         ascending: bool = False,
+        sort_by: list[str] | None = Check(default=None, examples=["price", "quantity"]),
+        is_disabled: bool | None = None,
```

Comment:
`is_disabled` should default to either a `True` or `False`, not `None`
```diff
- is_disabled: bool | None = None,
+ is_disabled: bool = False,
```

Reply:
If I want to get all the items, regardless of the `is_disabled` status? Say an Admin of the shop wants to show all the items the shop has had.

Reply:
`is_disabled` = True doesn't affect the fact that it show enabled item, it just additionally display disabled item

Reply:
If you want tri-state, it shouldn't be both `bool` and `None`, but instead Literal integer e.g. `0`, `1`, `2` or Literal String `display: Literal["enabled", "disabled", "both"] = "enabled"`

Reply:
> `is_disabled` = True doesn't affect the fact that it show enabled item, it just additionally display disabled item

Struggling to udnerstand this.

> If you want tri-state, it shouldn't be both `bool` and `None`, but instead Literal integer e.g. `0`, `1`, `2` or Literal String `display: Literal["enabled", "disabled", "both"] = "enabled"`

This is down to personal preference. Either way is valid, just different people prefer different methods.

Reply:
`is_disabled` is to divide whether to include disabled, not to only disabled I mean by

Multi-type is more difficult to handle and I am unsure how it would be handled in params (I wouldn't complain if it is in json data)